libs/payments/iap/src/lib/apple/apple-iap.service.ts

```diff
@@ -0,0 +1,168 @@
+
+    // STEP 2. Check if the purchase has been registered to a user. If it is, then return conflict error to our caller.
+    if (purchase.userId === userId) {
+      // Purchase record already registered to the target user. We'll do nothing.
+      return purchase;
+    } else if (purchase.userId) {
+      // this.log.info('Purchase already registered', { purchase });
+      // Purchase record already registered to different user. Return 'conflict' to caller
+      const libraryError = new AppleIapConflictError(
+        'Purchase has been registered to another user'
+      );
+      this.log.error('registerToUserAccount.PurchaseUpdateError.Conflict', {
+        err: libraryError,
+        originalTransactionId,
+      });
+      throw libraryError;
+    }
+
+    // STEP 3: Register purchase to the user
+    await this.appleIapPurchaseManager.forceRegisterToUser(
+      originalTransactionId,
+      userId
+    );
+
+    return purchase;
+  }
+
+  /**
+   * Update Firestore with the latest subscription status information from
+   * the App Store Server.
+   *
+   * See https://developer.apple.com/documentation/appstoreservernotifications/notificationtype
+   * and https://developer.apple.com/documentation/appstoreservernotifications/subtype
+   */
+  async processNotification(
+    bundleId: string,
+    originalTransactionId: string,
+    notification: DecodedNotificationPayload
+  ): Promise<AppStoreSubscriptionPurchase | null> {
+    // Type-guard to ensure the notification is for a subscription
+    if (
+      NOTIFICATION_TYPES_FOR_NON_SUBSCRIPTION_PURCHASES.includes(
+        notification.notificationType
+      )
+    ) {
+      return null;
+    }
+
+    // We can safely ignore Subscribed - InitialBuy notifications, because
+    // with a new subscription, our iOS app will send the same original
+    // transaction ID to the server for verification.
+    if (
+      notification.notificationType === NotificationType.Subscribed &&
+      notification?.subtype === NotificationSubtype.InitialBuy
+    ) {
+      return null;
+    }
+
+    // For other types of notifications, we query the App Store Server API
+    // to update our purchase record cache in Firestore
+    return this.querySubscriptionPurchase(
+      bundleId,
+      originalTransactionId,
+      notification.notificationType,
+      notification?.subtype
+    );
+  }
+
+  /**
+   * Retrieves a list of known subscription purchases for the customer.
+   */
+  async getSubscriptions(
+    userId: string
+  ): Promise<AppStoreSubscriptionPurchase[]> {
+    const allPurchases = await this.appleIapPurchaseManager.getForUser(userId);
+    const purchases = allPurchases.filter((purchase) =>
+      purchase.isEntitlementActive()
+    );
+
+    return purchases;
+    // return this.stripeHelper.addPriceInfoToIapPurchases(
+    //   purchases,
+    //   MozillaSubscriptionTypes.IAP_APPLE
+    // );
+  }
+
+  async querySubscriptionPurchase(
```

Comment:
This should be a manager-level method, not a service-level one. This should be removed imo.

See my comment here: https://mozilla.slack.com/archives/C04RBGF9HRN/p1742920172578469